src/sql/workbench/parts/query/browser/queryResultsView.ts

```diff
@@ -212,26 +213,24 @@ export class QueryResultsView extends Disposable {
 
 	private setQueryRunner(runner: QueryRunner) {
 		const activeTab = this._input.state.activeTab;
-		if (runner.hasCompleted && !this.hasResults(runner)) {
-			this.hideResults();
-		} else {
+		if (this.hasResults(runner)) {
 			this.showResults();
```

Comment:
So this change here is making it so new query tabs won't show their results until ran a second time. This is because by the time this is called we're already past the onQueryStart event being fired so the below handler never gets called which means we never call showResults to un-hide the results tab